prbt_hardware_support/test/integrationtests/integrationtest_stop1_service_missing.cpp

```diff
@@ -0,0 +1,147 @@
+ */
+class Stop1ServiceMissingIntegrationTest : public ::testing::Test
+{
+public:
+  /**
+   * @brief Gather service-advertising functions for all manipulator services and determine service to omit.
+   */
+  void SetUp() override;
+
+protected:
+  ros::NodeHandle nh_;
+  ros::NodeHandle nh_priv_ {"~"};
+
+  ros::ServiceServer hold_server_;
+  ros::ServiceServer unhold_server_;
+  ros::ServiceServer halt_server_;
+  ros::ServiceServer recover_server_;
+
+  std::function<void()> omit_advertise_func_;
+  //! Map holding all but one service-advertising functions
+  std::map<std::string, std::function<void()>> advertise_funcs_;
+};
+
+void Stop1ServiceMissingIntegrationTest::SetUp()
+{
+  ROS_DEBUG("SetUp()");
+
+  advertise_funcs_[RECOVER_SERVICE_NAME] = [this](){
+    this->recover_server_ = this->nh_.advertiseService(RECOVER_SERVICE_NAME, triggerCallbackDummy); };
+  advertise_funcs_[HALT_SERVICE_NAME] = [this](){
+    this->halt_server_ = this->nh_.advertiseService(HALT_SERVICE_NAME, triggerCallbackDummy); };
+  advertise_funcs_[UNHOLD_SERVICE_NAME] = [this](){
+    this->unhold_server_ = this->nh_.advertiseService(UNHOLD_SERVICE_NAME, triggerCallbackDummy); };
+  advertise_funcs_[HOLD_SERVICE_NAME] = [this](){
+    this->hold_server_ = this->nh_.advertiseService(HOLD_SERVICE_NAME, triggerCallbackDummy); };
+
+  std::string omit_service;
+  ASSERT_TRUE(nh_priv_.getParam(OMIT_SERVICE_PARAM_NAME, omit_service));
+
+  const auto &it = advertise_funcs_.find(omit_service);
+  ASSERT_TRUE(it != advertise_funcs_.end()) << "Invalid service name: " << omit_service;
+
+  omit_advertise_func_ = it->second;
+  advertise_funcs_.erase(it);
+}
+
+/**
+ * @brief Test that the startup of the stop1_executor is not complete if a service is missing.
+ *
+ * @tests{stop1_missing_services,
+ *  Test that the startup of the stop1_executor is not complete if a service is missing.
+ * }
+ *
+ * Test Sequence:
+ *    1. Advertise all but one manipulator services
+ *    2. Advertise missing service from step 1
+ *
+ * Expected Results:
+ *    1. Stop1 executor node does not come up
+ *    2. Stop1 executor node does complete startup
+ */
+TEST_F(Stop1ServiceMissingIntegrationTest, testMissingService)
+{
+  /**********
+   * Step 1 *
+   **********/
+  ROS_DEBUG("Step 1");
+
+  std::for_each(advertise_funcs_.begin(),
+                advertise_funcs_.end(),
+                [](const std::pair<std::string, std::function<void()>> &el){ el.second(); });
+
+  ros::Duration(WAIT_FOR_NODE_SLEEPTIME_S).sleep();
+  EXPECT_FALSE(checkForNode(STOP1_EXECUTOR_NODE_NAME));
+
+  /**********
+   * Step 2 *
+   **********/
+  ROS_DEBUG("Step 2");
+
+  omit_advertise_func_();
+  waitForNode(STOP1_EXECUTOR_NODE_NAME);
+}
+
+} // namespace prbt_hardware_support
+
+
+int main(int argc, char *argv[])
+{
+  ros::init(argc, argv, "integrationtest_stop1_service_missing");
+  ros::NodeHandle nh;
+
+  ros::AsyncSpinner spinner{1};
+  spinner.start();
```

Comment:
Needed?

Reply:
Yes I think. `ros::spin()` would block the test execution.